lc-hax/Scripts/Helpers/ShipDoor.cs

```diff
@@ -0,0 +1,19 @@
+using System.Linq;
+using UnityEngine;
+
+namespace Hax;
+
+public static partial class Helper {
+    public static InteractTrigger? GetInteractTriggerForAnimation(GameObject? gameObject, string animation) {
+        return gameObject.Unfake()?
+            .GetComponentsInChildren<AnimatedObjectTrigger>()
+            .FirstOrDefault(trigger => trigger.animationString == animation)?
+            .GetComponentInParent<InteractTrigger>();
+    }
+
+    public static void CloseShipDoor(bool closed) {
+        if (!Object.FindObjectOfType<HangarShipDoor>().IsNotNull(out HangarShipDoor shipDoor)) return;
+
+        GetInteractTriggerForAnimation(shipDoor.gameObject, closed ? "CloseDoor" : "OpenDoor")?.onInteract.Invoke(Helper.LocalPlayer);
```

Comment:
Why not just use the `Interact` method directly?

```suggestion
        GetInteractTriggerForAnimation(shipDoor.gameObject, closed ? "CloseDoor" : "OpenDoor")?.Interact(Helper.LocalPlayer.transform);
```

Reply:
`Interact` also deals with player animations regarding object usage, e.g. the ladder anim, whereas `Invoke` only affects the object itself

Reply:
I see, so it's better to use `onInteract` to skip the overhead.